### PR TITLE
Add CI for cocotb_caravel 

### PR DESCRIPTION
**Pull request by M0stafaRady**

### CODE REVIEW

File: .github/workflows/caravel_cocotb.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Caravel Cocotb CI
+
+on:
+  push:
+    branches:
```
Comment (marwaneltoukhy) [inline .github/workflows/caravel_cocotb.yml:5]: You can use '*' for all branches

File: .github/workflows/caravel_cocotb.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - '.github/workflows/caravel_cocotb.yml'
+      - 'cocotb/caravel_cocotb/scripts/**'
+      - '!cocotb/caravel_cocotb/**/'  # Exclude directories within the specified directory
+
+
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (marwaneltoukhy) [inline .github/workflows/caravel_cocotb.yml:20]: You can create a matrix where it tries out several python versions. (This is a future improvement)

**Review by marwaneltoukhy — COMMENTED** [review]
Have you tried to get something to fail?

**Comment by M0stafaRady** [comment]
> Have you tried to get something to fail?

yes I did.

**Pull request merged**